graphql/resolvers/search.ts

```
@@ -52,8 +93,20 @@ const resolvers = {
         )
       );
 
+      const termInfo: TermInfo = await prisma.termInfo.findFirst({
+        where: { termId: "" + args.termId },
+      });
+      let maxEndDate = -1;
```

This should throw an error. You've just executed a search, so you know that the term ID exists. If it doesn't exist for some reason, that should immediately throw an error.